moabb/evaluations/splitters.py

```diff
@@ -101,3 +111,138 @@ def split(self, y, metadata):
+    """
+
+    def __init__(
+        self,
+        cv_class: type[BaseCrossValidator] = LeaveOneGroupOut,
+        shuffle: bool = False,
+        random_state: int = None,
+        **cv_kwargs,
+    ):
+        self.cv_class = cv_class
+        self.cv_kwargs = cv_kwargs
+        self._cv_kwargs = dict(**cv_kwargs)
+
+        self.shuffle = shuffle
+        self.random_state = random_state
+
+        self._rng = check_random_state(random_state) if shuffle else None
+
+        if not shuffle and random_state is not None:
+            raise ValueError("`random_state` should be None when `shuffle` is False")
+
+        if shuffle and len(self.cv_kwargs) == 0 and cv_class is LeaveOneGroupOut:
+            log.warning(
+                "Shuffling is not implemented for LeaveOneGroupOut. "
+                "The `shuffle` parameter change the behaviour of the splitter."
+                "Use GroupShuffleSplit instead."
+            )
+
+            self.cv_class = GroupShuffleSplit
+
+        params = inspect.signature(self.cv_class).parameters
+        for p, v in [
+            ("shuffle", shuffle),
+            ("random_state", self._rng),
+        ]:
+            if p in params:
+                self._cv_kwargs[p] = v
+
+    def get_n_splits(self, metadata):
+        if self.cv_class is LeaveOneGroupOut:
+            return metadata.groupby(["subject", "session"]).ngroups
+        else:
+            return self.cv_class(**self._cv_kwargs).get_n_splits(metadata)
+
+    def split(self, y, metadata):
+        # here, I am getting the index across all the subject
+        all_index = metadata.index.values
+        # I check how many subjects are here:
+        subjects = metadata["subject"].unique()
+
+        # For the subject that is shuffle now, I am getting the subject index
+        for subject in subjects:
+            # Creating the subject_mask
+            subject_mask = metadata["subject"] == subject
+            # from all the index, I am getting the trial index
+            subject_indices = all_index[subject_mask]
+            # Here, I am getting the metainformation to use the column session
+            subject_metadata = metadata[subject_mask]
+            # getting the label at subject level
+            y_subject = y[subject_mask]
+            # check the number of sessions and check how many sessions we
+            # have!
+            sessions = subject_metadata["session"].unique()
+
+            if len(sessions) <= 1:
+                log.info(
+                    f"Skipping subject {subject}: Only one session available"
+                    f"Cross-session evaluation requires at least two sessions."
+                )
+                continue  # Skip subjects with only one session
+
+            # if shuffle is True, I am shuffling the session with group shuffle
+            if self.shuffle:
+                if self._cv_kwargs.get("n_splits") is None:
+                    self._cv_kwargs["n_splits"] = len(sessions)
```

Comment:
Why do you need this step?
The group shuffle split has a default value that should be respected

Reply:
I was doing this step because I was trying to incorporate the shuffle into the LeaveOneGroupOut.